bot/rules/duplicates.py

```diff
@@ -13,6 +13,7 @@ async def apply(
         if (
             msg.author == last_message.author
             and msg.content == last_message.content
+            and (msg.content and not msg.attachments)
```

Comment:
The attachments filter checks whether too many attachments are sent. This isn't mutually exclusive with duplicate messages, as someone might still send the same text but with a message attached. So it seems to me like this should just check that there is any text at all, and not skip messages with attachments.